src/components/Timeout/Timeout.js

```diff
@@ -0,0 +1,203 @@
+  },
+  modal: {
+    position: 'relative',
+    background: 'white',
+    width: '38rem',
+    height: '28rem',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    margin: '28vh auto',
+    '@media (max-width: 600px)': {
+      width: '92vw',
+      height: '45vh',
+    },
+  },
+  modal__center_div: {
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'space-between',
+  },
+  modal__center_div__content: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  modal__center_div__content__p_head: {
+    fontSize: '24px',
+    fontWeight: '700',
+    margin: '0',
+    textAlign: 'center',
+    color: 'rgb(80, 80, 80)',
+  },
+  modal__center_div__content__p_timer: {
+    fontSize: '21px',
+    fontWeight: '600',
+    textAlign: 'center',
+    color: 'rgb(0, 98, 255)',
+  },
+  modal__center_div__content__p_text: {
+    textAlign: 'center',
+    color: 'rgb(80, 80, 80)',
+  },
+  modal__center_div__buttons: {
+    display: 'flex',
+    justifyContent: 'center',
+    marginTop: '1.5rem',
+  },
+  modal__center_div__buttons__button_logoff: {
+    padding: '.5rem 1rem',
+    fontWeight: '700',
+    textTransform: 'uppercase',
+    color: 'white',
+    borderRadius: '3px',
+    transition: '200ms ease-in-out',
+    cursor: 'pointer',
+    background: 'gray',
+    marginRight: '1rem',
+    '&:hover': {
+      filter: 'brightness(1.2)',
+      transition: '300ms ease-in-out',
+    },
+  },
+  modal__center_div__buttons__button_continue: {
+    padding: '.5rem 1rem',
+    fontWeight: '700',
+    textTransform: 'uppercase',
+    color: 'white',
+    borderRadius: '3px',
+    transition: '200ms ease-in-out',
+    cursor: 'pointer',
+    background: 'rgb(56, 132, 255)',
+    marginLeft: '1rem',
+    '&:hover': {
+      filter: 'brightness(1.2)',
+      transition: '300ms ease-in-out',
+    },
+  },
+});
+
+export default function Timeout(props) {
+  //track idle time in background
+  const idleTime = parseInt(process.env.REACT_APP_IDLETIME);
+  const countdownTime = parseInt(process.env.REACT_APP_COUNTDOWNTIME);
+
+  const { logoff, timedOut } = Auth;
+  const isLoggedIn = Auth.isLoggedIn();
+
+  const [idle, setIdle] = useState(false);
+  const time = useRef(0);
+
+  function reset() {
+    time.current = 0;
+  }
+
+  useEffect(() => {
+    let interval;
+    if (isLoggedIn && !idle) {
+      interval = setInterval(() => {
+        time.current += 1000;
+        if (time.current === idleTime) {
+          setIdle(true);
+          clearInterval(interval);
+        }
+      }, 1000);
+    }
+
+    return () => {
+      if (interval) clearInterval(interval);
+    };
+  }, [idleTime, idle, isLoggedIn]);
+
+  return (
+    <div onMouseOver={() => reset()} onKeyPress={() => reset()}>
+      {idle && (
+        <Modal
+          countdownTime={countdownTime}
+          setIdle={setIdle}
+          logoff={logoff}
+          timedOut={timedOut}
+        />
+      )}
+      {props.children}
+    </div>
```

Comment:
refactor, if you have more than 2 or more attribs use es6 syntax:

`<div onMouseOver={reset} onKeyPress={reset}>
      {idle && (
        <Modal
          {...{countdownTime: countdownTime,
          setIdle: setIdle,
          logoff: logoff,
          timedOut: timedOut}}
        />)}{children}</div>`